src/generator.cs

```diff
@@ -5210,14 +5210,21 @@ void PrintExport (MemberInformation minfo)
 			print ("[Export (\"{0}\"{1})]", minfo.selector, minfo.is_variadic ? ", IsVariadic = true" : string.Empty);
 	}
 
-	void PrintExport (ExportAttribute ea)
+	void PrintExport (MemberInformation minfo, ExportAttribute ea)
 	{
-		PrintExport (ea.Selector, ea.ArgumentSemantic);
+		PrintExport (minfo, ea.Selector, ea.ArgumentSemantic);
 	}
 
-	void PrintExport (string sel, ArgumentSemantic semantic)
+	void PrintExport (MemberInformation minfo, string sel, ArgumentSemantic semantic)
 	{
-		if (semantic != ArgumentSemantic.None)
+		bool output_semantics = semantic != ArgumentSemantic.None;
+		// it does not make sense on every properties, depending on the their types
+		if (output_semantics && (minfo.mi is PropertyInfo)) {
+			var t = minfo.property.PropertyType;
+			output_semantics = t.IsPrimitive ? t == typeof (IntPtr) : t != typeof (string);
```

Comment:
Looks like this won't print ArgumentSemantic for strings, but afaict strings (NSString) can have different ArgumentSemantic.